test/server/test-run-test.js

```diff
@@ -0,0 +1,248 @@
+            testRunDone = true;
+        });
+
+        return testRun[CLIENT_MESSAGES.ready]()
+            .then(function (command) {
+                expect(testRun.errs[0]).eql(testError);
+                expect(command.type).eql(commands.TYPE.testDone);
+                expect(testRunDone).to.be.true;
+            });
+    });
+
+    it('Should execute the wait command without sending request to the client', function () {
+        var waitingStartedTime = Date.now();
+
+        var testMock = {
+            fixture: 'fixture',
+            fn:      function (testRun) {
+                return testRun
+                    .executeCommand(commands.createWaitCommand(50))
+                    .then(function () {
+                        expect(Date.now() - waitingStartedTime).to.be.least(50);
+                        testRun._done();
+                    });
+            }
+        };
+
+        var testRun = new TestRun(testMock);
+
+        return testRun[CLIENT_MESSAGES.ready]()
+            .then(function (command) {
+                expect(command.type).eql(commands.TYPE.testDone);
+                expect(Date.now() - waitingStartedTime).to.be.least(50);
+                expect(testRun.errs.length).eql(0);
+            });
+    });
+
+    it('Should raise an error when try to execute a new command while the previous on is not finished', function () {
+        var testMock = {
+            fixture: 'fixture',
+            fn:      noop
+        };
+
+        var testRun = new TestRun(testMock);
+
+        testRun.executeCommand(commands.createClickCommand());
+
+        return testRun.executeCommand(commands.createClickCommand())
+            .then(function () {
+                throw 'Command should be rejected';
+            })
+            .catch(function (err) {
+                expect(err.toString()).eql('Error: Attempt to execute a command when the previous one is still executing is detected');
+            });
+    });
+
+    it('Should reject pending command if an js-error is raised', function () {
+        var testMock = {
+            fixture: 'fixture',
+            fn:      noop
+        };
+
+        var clickCommand = commands.createClickCommand();
+        var jsError      = 'js-error';
+
+        var testRun = new TestRun(testMock);
+
+        nextTick()
+            .then(function () {
+                return testRun[CLIENT_MESSAGES.ready]();
+            })
+            .then(function () {
+                testRun[CLIENT_MESSAGES.jsError]({ type: 'failed', err: jsError });
+            });
+
+        return testRun
+            .executeCommand(clickCommand)
+            .then(function () {
+                throw 'command executing should fail but it is succeed';
+            })
+            .catch(function (err) {
+                expect(err).eql(jsError);
+            });
+    });
+
+    it('Should reject the next command if an js-error is raised when there is no pending command', function () {
```

Comment:
Should reject the next command if there are no pending commands after a js-error is raised.